### PR TITLE
Feature/unit testing

### PR DESCRIPTION
**Pull request by niketansainiantier**

### CODE REVIEW

File: contracts/ERC20/EndToken.sol [hunk 1]
```diff
@@ -218,9 +181,23 @@ contract EndToken is IEndToken, ERC20Upgradeable, AccessControlUpgradeable {
...
+        yearlyVestAmount[time / 31536000] = VestAmount(mintAmount, true, true, true);
+        mint(address(this), mintAmount);
+        if (mintFee != 100) {
+            mintFee -= 100;
+        }
+        lastYear = time / 31536000;
+        emit MintAndVest(block.timestamp, mintAmount);
+        }
+    }
+    function _transfer(address from, address to, uint256 amount) internal override {
```
Comment (Beardev118) [inline contracts/ERC20/EndToken.sol:198]: Please fix this _transfer function overriding
In the Openzeppelin 5.0 or above, we can't override this _transfer function

Please check other contracts such as StEth.sol

File: contracts/ERC20/mockStEth.sol [hunk 2]
```diff
@@ -6,6 +6,7 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 // import "@aragon/os/contracts/lib/math/SafeMath.sol";
 // import "./utils/Pausable.sol";
 
+
```
Comment (Beardev118) [inline contracts/ERC20/mockStEth.sol:9]: RM this empty line

File: contracts/EnderBondLiquidityDeposit.sol [hunk 3]
```diff
@@ -7,6 +7,7 @@ import "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 import "contracts/interfaces/ISTETH.sol";
+import "hardhat/console.sol";
```
Comment (Beardev118) [inline contracts/EnderBondLiquidityDeposit.sol:10]: Why did you add this line?
RM this line, you didn't use it

**Comment by Beardev118** [comment]
[SC-003 Smart contract PR review](https://trello.com/c/DluBGZ7t/5-sc-003-smart-contract-pr-review)

File: contracts/EnderBond.sol [hunk 4]
```diff
@@ -384,14 +387,18 @@ event ClaimRewards(address indexed account, uint256 reward,uint256 tokenId);
...
         signData memory userSign
     ) public payable nonReentrant depositEnabled bondPaused returns (uint256 tokenId) {
-        if (principal < minDepositAmount) revert InvalidAmount();
-        if (maturity < 5 || maturity > 90) revert InvalidMaturity();
-        if (token != address(0) && !bondableTokens[token]) revert NotBondableToken();
-        if (bondFee > 10000) revert InvalidBondFee();
-        if(isWhitelisted){
-            address signAddress = _verify(userSign);
-            if(signAddress != signer || userSign.user != msg.sender) revert NotWhitelisted();
+        address _owner = owner();
```
Comment (Beardev118) [inline contracts/EnderBond.sol:390]: Please check the owner() function

Comment (niketansainiantier) [reply contracts/EnderBond.sol:390]: I'm not getting your point. Please explain what you want to say.

Here owner function returns the address of the owner.

File: contracts/EnderBondLiquidityDeposit.sol (hunk 3 above)
Comment (niketansainiantier) [reply contracts/EnderBondLiquidityDeposit.sol:10]: Since the contract has already been deployed on the mainnet, it seems prudent not to expend further time on this matter.

File: contracts/ERC20/mockStEth.sol (hunk 2 above)
Comment (niketansainiantier) [reply contracts/ERC20/mockStEth.sol:9]: Why are we investing time in this mock contract when it's not a real one?

File: contracts/ERC20/EndToken.sol (hunk 1 above)
Comment (niketansainiantier) [reply contracts/ERC20/EndToken.sol:198]: we were not using the 5.0 version. At the time of initial development 5.0 stable version was not there.

And I guess, it's fine to use the last updated 4.9.2 or its latest version. As per the versioning mechanism. because we don't want to add any breaking changes to it.

File: contracts/EnderBondLiquidityDeposit.sol (hunk 3 above)
Comment (Beardev118) [reply contracts/EnderBondLiquidityDeposit.sol:10]: Without testing, how did you deploy the smart contract to the mainnet?
did you discuss with Winnal?

File: contracts/EnderBond.sol (hunk 4 above)
Comment (Beardev118) [reply contracts/EnderBond.sol:390]: did you compile the contract in your side? I I got the error for that

Comment (ravipal20) [reply contracts/EnderBond.sol:390]: yes, we've compile the contract at our side. Error is coming because you guys made lot of changes in the package.json file. At our end it works fine
Screenshot from 2024-03-04 15-32-02

File: contracts/EnderBondLiquidityDeposit.sol (hunk 3 above)
Comment (Beardev118) [reply contracts/EnderBondLiquidityDeposit.sol:10]: Please review this
https://github.com/enderprotocol/ender-v1/blob/feature/unitTesting/test/unit/EnderBond.test.js

Comment (Beardev118) [reply contracts/EnderBondLiquidityDeposit.sol:10]: I can't find the test file for EnderBond contract

Comment (ravipal20) [reply contracts/EnderBondLiquidityDeposit.sol:10]: Please check the EnderBondDepositAndWithdraw.test.js file for ender bond and for EnderBondLiquidityDeposit contract check the EnderBondLiquidityDeposit.test.js file

**Review by Beardev118 — APPROVED** [review]
